src/Test/Inspection/Core.hs

```diff
@@ -203,43 +205,56 @@ eqSlice it slice1 slice2
 -- | Returns @True@ if the given core expression mentions no type constructor
 -- anywhere that has the given name.
 freeOfType :: Slice -> [Name] -> Maybe (Var, CoreExpr)
-freeOfType slice tcNs = allTyCons (\tc -> getName tc `notElem` tcNs) slice
-
-allTyCons :: (TyCon -> Bool) -> Slice -> Maybe (Var, CoreExpr)
-allTyCons predicate slice = listToMaybe [ (v,e) | (v,e) <- slice, not (go e) ]
+freeOfType slice tcNs =
+      fmap (\(a,b,_) -> (a,b))
+    $ allTyCons (\tc -> getName tc `notElem` tcNs) slice
+
+allTyCons :: (TyCon -> Bool) -> Slice -> Maybe (Var, CoreExpr, TyCon)
```

Comment:
Might you add a comment about what the function does?

Reply:
Will do.